Backend/WebAPI/Controllers/ResetController.cs

```diff
@@ -0,0 +1,123 @@
+                                //Give the user a new resetID
+                                string resetID = rs.CreateResetID();
+                                rs.AddResetIDToDB(email, resetID);
+                                string resetURL = rs.CreateResetURL(resetID);
+                                rs.SendResetEmailUserExists(email, resetID);
+                            }
+                            else
+                            {
+                                //Tell the user that to check their email
+                            }
+                        }
+                        else
+                        {
+                            //get the expiration date of the token, and add 24 hours to it
+                            //if 24 hours has passed since they failed the security questions
+                                //delete all tokens
+                            //else 
+                                //Tell the user they cannot attempt to reset the password for 24 hours
+                        }
+                    }
+                    else
+                    {
+                        // get the most recent token
+                        //get the expiration time for that token
+                        //add 24 hours to it
+                        //check if that time is after the current time
+                        //if the 24 hours hasn't passed
+
+                            //Tell the user they cannot attempt to reset the password for 24 hours
+                        //if the 24 hours has passed
+                            //delete all tokens
+                    }
+                }
+            }
+            else
+            {
+                rs.SendResetEmailUserDoesNotExist(email);
+            }
+        }
+
+        //After the user clicks the link in the email, this action gets called
+        [HttpPost]
+        [ActionName("ResetPassword")]
+        public void ResetPassword(string resetID)
+        {
+            ResetService rs = new ResetService();
+            //Check to see if the resetID has been locked out 
+            if (!rs.IsLockedOut(resetID)) //ResetID is still valid
+            {
+                if (!rs.IsResetIDIsExpired(resetID))
+                {
+                    //Check to see how many times they've attempted 
+                    if (rs.GetResetCount(resetID) == 3)
+                    {
+                        rs.LockResetID(resetID); //If the user has attempted to reset the password 3 times already with the same resetID, lock that resetID
+                    }
+                    else
+                    {
+                        //Get the security questions from the DB
+                        rs.GetSecurityQuestionsFromDB(resetID);
+                    }
+                }
+                else
+                {
+                    rs.DeleteResetIDFromDB(resetID);
```

Comment:
Do not do this. Simply inform the user that the link has expired and redirect to the reset page. The token has expired in the DB, no need to delete it.